### ISSUE TITLE
`stack haddock` doesn’t include sources for boot libraries

### ISSUE THREAD

**Issue by lexi-lambda** [commercialhaskell/stack#4985]
### General summary/comments (optional)

The haddock documentation generated by `stack build --haddock` does not appear to include sources for boot libraries. This means that there is no per-module “Source” link in the upper right navigation bar (it usually appears between the “Quick Jump” and “Contents” links), and individual bindings do not have “Source” links, either.

This seems like it might be because GHC itself ships with documentation for the boot libraries prebuilt, so maybe this is really an upstream issue. But I wonder if it’s possible to get stack to regenerate the documentation for those packages, or if they’re totally off-limits and stack won’t ever rebuild the haddocks for them.

### Steps to reproduce

  1. Run `stack haddock --open <PKG>`, where `<PKG>` is a boot package like `base`, `process`, or `transformers`.

  2. Click on any module link, such as `Control.Applicative`, `System.Process`, or `Control.Monad.Trans.Reader`.

  3. Note that there are no “Source” links present.

  4. Do the same for a non-boot package, such as `hashable`, and notice that the “Source” links *are* present, as they ought to be.

### Stack / OS version

```
$ stack --version
Version 2.1.3, Git revision 0fa51b9925decd937e4a993ad90cb686f88fa282 (7739 commits) x86_64 hpack-0.31.2
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.14.5
BuildVersion:	18F132
```

Method of installation: `stack upgrade`.

**Comment by borsboom** [comment]
> This seems like it might be because GHC itself ships with documentation for the boot libraries prebuilt, so maybe this is really an upstream issue.

This is correct, and those prebuilt haddocks don't include the source links.

> But I wonder if it’s possible to get stack to regenerate the documentation for those packages, or if they’re totally off-limits and stack won’t ever rebuild the haddocks for them.

I'm less knowledgeable about this area, but basically Stack does not rebuild the boot libraries (doing so causes too many problems), and AFAIK Cabal won't build haddocks without also building the package, so this isn't trivial to change.

That said, in theory Stack could build the package in a different location just for haddocks and ignore the built library, but this would introduce a weird code path that I'm not sure is worth it.

It would be worth inquiring whether the GHC team would consider including source links with their haddocks.  They may well have good reasons not to, of course (such as extra disk space usage).

**Comment by mpilgrem** [comment]
I am closing this issue because with the current version of Stack and LTS 22.13, `stack haddock` does appear to produce documentation that links to the source code of the `base` package.
